Feature/add react router

abdulqshabbir opened the pull request:
Added react router to our application.

`PrivateRoute` component wraps any components that are protected and require authentication. If user is not authenticated they redirect the user to the login page.

`Helmet` component can be used to wrap any page-level components with helmet

`SideBar` has been lifted up into a `WithSideBar` component that you can use to render the sidebar next to any other page-level component of your choosing.

amirfakhrullah committed:
setup login signup and logout

amirfakhrullah committed:
swagger init

amirfakhrullah committed:
update readme with the api docs link

amirfakhrullah committed:
/api/auth/check endpoint

amirfakhrullah committed:
proper response

amirfakhrullah committed:
reformat

amirfakhrullah committed:
resolve merge conflicts

amirfakhrullah committed:
fix folder structures

amirfakhrullah committed:
refactor

amirfakhrullah committed:
reformat with prettier --write

amirfakhrullah committed:
remove prettier-quick

amirfakhrullah committed:
remove comments

abdulqshabbir committed:
added react-router-dom v6 as dependency

abdulqshabbir committed:
added react-router-dom v6 as dependency

abdulqshabbir committed:
added PrivateRoute component which redirects to login page if user is not authed

abdulqshabbir committed:
Created helmet wrapper component

abdulqshabbir committed:
Created helmet wrapper component

abdulqshabbir committed:
created a WithSideBar component that renders one component next to the sidebar

abdulqshabbir committed:
created a WithSideBar component that renders one component next to the sidebar

abdulqshabbir committed:
Renamed main exports to index.tsx to follow front end convention

abdulqshabbir committed:
Renamed main exports to index.tsx to follow front end convention

amirfakhrullah commented on client/src/components/Helmet/index.tsx:
```diff
@@ -1,8 +1,11 @@
+ return (
     <HelmetProvider>
       <Helmet>
         <title>Notum</title>
```
good one. Maybe can set the component to receive `title` prop so that the title can be dynamic and changeable based on pages

abdulqshabbir replied:
OK good point!

amirfakhrullah commented on client/src/components/Auth/PrivateRoute.tsx:
```diff
@@ -0,0 +1,22 @@
+    return <Navigate to="/login" state={{ from: location }} replace />;
+  }
+
+  return children;
```
Since all the private routes will have the sidebar, better to put the `WithSideBar` component in the `PrivateRoute` rather than putting `WithSideBar` components in all the private pages. It would look something like this:
```suggestion
  return (
     <ContainerStyles>
            <SideBar />
            {children}
     </ContainerStyles>
  );
```

amirfakhrullah replied:
`HelmetWrapper` component also might be better in here rather than in each of the private pages

abdulqshabbir replied:
> Since all the private routes will have the sidebar, better to put the `WithSideBar` component in the `PrivateRoute` rather than putting `WithSideBar` components in all the private pages. It would look something like this:

Wonderful suggestion :)

abdulqshabbir replied:
> `HelmetWrapper` component also might be better in here rather than in each of the private pages

If we do this then we have to remember to use the `HelmetWrapper` for each of the public routes. Is this OK?

cherylli replied:
Maybe something like a <Layout> component or whatever you want to call it which will include the helmet and other stuff we need in every page. I think we might need more than just helmet on every page

abdulqshabbir replied:
I think we should wait until we need an abstraction for creating that personally. I think abstracting too early can make the wrapping component inflexible and hard to maintain. Maybe we all take a vote on this?

amirfakhrullah replied:
@abdulqshabbir @cherylli 

Suggestion:
`PrivateRoutes` - have `HelmetWrapper`, `Sidebar` and routing to calendar, settings and search pages
`PublicRoutes` - have `HelmetWrapper` and routing to login and signup pages only

We have to check that if the user is logged in, they should not be able to access login and signup pages too, so I suggest `PublicRoutes` to check that if user is logged in, redirect them to calendar page (similar redirect pattern as `PrivateRoutes`)

amirfakhrullah replied:
Or if the `HelmetWrapper` is the same for all pages, we better put it on the top level: App.tsx

amirfakhrullah commented:
Overall looks good

amirfakhrullah committed:
implement validateRequestBody

12cassie34 reviewed (commented):
Looks good to me

abdulqshabbir committed:
Added title prop to HelmetWrapper

abdulqshabbir committed:
Added title prop to HelmetWrapper

abdulqshabbir committed:
PrivateRoute component renders SideBar

abdulqshabbir committed:
PrivateRoute component renders SideBar

amirfakhrullah committed:
helpers to calculate and arrang lessons for the week

amirfakhrullah committed:
reformat

amirfakhrullah committed:
remove unused stuffs

amirfakhrullah committed:
Merge pull request #45 from chingu-voyages/feature/auth

Feature/auth

cherylli reviewed (dismissed):
looks good

abdulqshabbir committed:
Added Public route
Also, Helmet wraps the entire application now and we can change this at a later time if we need the flexibilty

abdulqshabbir committed:
Added Public route
Also, Helmet wraps the entire application now and we can change this at a later time if we need the flexibilty

abdulqshabbir committed:
resolved merge conflict

abdulqshabbir committed:
Merge branch 'feature/add-react-router' of https://github.com/chingu-voyages/v40-bears-team-37 into feature/add-react-router
